lib/overcommit/hook_runner.rb

```diff
@@ -74,7 +74,13 @@ def run_hooks
 
         print_results
 
-        !(@failed || @interrupted)
+        hook_failed = @failed || @interrupted
+
+        if hook_failed && @context.respond_to?(:post_fail_message)
```

Comment:
I thought it was best to allow such a method to not be implemented on the individual `HookContext` subclasses, but work here if it was. Right now only `HookContext::CommitMsg` implements it, to solve #494.